src/deploy/functions/prepare.ts

```diff
@@ -80,7 +80,15 @@ export async function prepare(
       clc.bold(options.config.src.functions.source) +
       " directory for uploading..."
   );
-  context.functionsSource = await prepareFunctionsUpload(runtimeConfig, options);
+  if (wantBackend.cloudFunctions.find((fn) => fn.platform === "gcfv1")) {
+    context.functionsSourceV1 = await prepareFunctionsUpload(runtimeConfig, options);
+  }
+  if (wantBackend.cloudFunctions.find((fn) => fn.platform === "gcfv2")) {
+    context.functionsSourceV2 = await prepareFunctionsUpload(
```

Comment:
Is there an opportunity for us to inform the user that runtimeconfig will be ignored for GCFv2 deploy?

Reply:
I don't think so? It's perfectly valid, though wonky, to use config in your v1 functions and not in your v2 functions. And I don't recall your migrate script deleting config either. Do you have a good way to decide when to warn?